### PR TITLE
Fix for WFCORE-923

### PR DESCRIPTION
**Pull request by jfdenise**
A new command to get/set/reset the command timeout. XML+option to provide a timeout via configuration. Actual implementation of timeout for low level operation and CommandHandlers.

### CODE REVIEW

**Comment by jfdenise** [comment]
retest this please!

**Comment by jfdenise** [comment]
retest this please!

**Comment by jfdenise** [comment]
retest this please!

**Comment by jfdenise** [comment]
retest this please!

**Comment by bstansberry** [comment]
retest this please

**Comment by bstansberry** [comment]
retest this please

File: testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CommandTimeoutHandlerTestCase.java [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+import static org.junit.Assert.assertEquals;
+import org.wildfly.core.testrunner.ServerControl;
+import org.wildfly.core.testrunner.ServerController;
+
+/**
+ *
+ * @author jdenise@redhat.com
+ */
+@RunWith(WildflyTestRunner.class)
+@ServerControl(manual = true)
```
Comment (bstansberry) [inline testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CommandTimeoutHandlerTestCase.java]: @jfdenise I believe this needs to be in testsuite/manualmode as the tests in testsuite/standalone are written assuming the server gets launched at the start of the surefire-plugin execution and then remains running through the entire set of tests in that execution. Manually controlling the server in the middle can interfere with this.

**Comment by jfdenise** [comment]
retest this please

**Pull request merged**